src/hipscat/io/write_metadata.py

```diff
@@ -0,0 +1,112 @@
+
+
+def write_catalog_info(args, histogram):
+    """Write a catalog_info.json file with catalog metadata
+
+    Args:
+        args (:obj:`PartitionArguments`): collection of runtime arguments for the partitioning job
+        histogram (:obj:`np.array`): one-dimensional numpy array of long integers where the
+            value at each index corresponds to the number of objects found at the healpix pixel.
+    """
+    metadata = {}
+    metadata["catalog_name"] = args.catalog_name
+    metadata["version"] = pkg_resources.get_distribution("hipscat").version
+    now = datetime.now()
+    metadata["generation_date"] = now.strftime("%Y.%m.%d")
+    metadata["ra_kw"] = args.ra_column
+    metadata["dec_kw"] = args.dec_column
+    metadata["id_kw"] = args.id_column
+    metadata["total_objects"] = histogram.sum()
+
+    metadata["origin_healpix_order"] = args.highest_healpix_order
+    metadata["pixel_threshold"] = args.pixel_threshold
+
+    metadata_filename = os.path.join(args.catalog_path, "catalog_info.json")
+    write_json_file(metadata, metadata_filename)
+
+
+def write_partition_info(args, destination_pixel_map):
+    """Write all partition data to CSV file.
+    Args:
+        args (:obj:`PartitionArguments`): collection of runtime arguments for the partitioning job
+        destination_pixel_map (:obj:`DataFrame`)
+          data frame that has as columns:
+          - pixel order of destination
+          - pixel number of destination
+          - sum of rows in destination
+          - list of all source pixels at original order
+    """
+    metadata_filename = os.path.join(args.catalog_path, "partition_info.csv")
+    data_frame = pd.DataFrame(destination_pixel_map.keys())
+    data_frame.columns = ["order", "pixel", "num_objects"]
+    data_frame.to_csv(metadata_filename, index=False)
+
+
+def write_legacy_metadata(args, histogram, pixel_map):
+    """Write a <catalog_name>_meta.json with the format expected by the prototype catalog.
+
+    Args:
+        args (:obj:`PartitionArguments`): collection of runtime arguments for the partitioning job
+        histogram (:obj:`np.array`): one-dimensional numpy array of long integers where the
+            value at each index corresponds to the number of objects found at the healpix pixel.
+        pixel_map (:obj:`np.array`): one-dimensional numpy array of integer 3-tuples.
+            See `pixel_math.partition_stats.generate_alignment` for more details on this format.
+    """
+    metadata = {}
+    metadata["cat_name"] = args.catalog_name
+    metadata["ra_kw"] = args.ra_column
+    metadata["dec_kw"] = args.dec_column
+    metadata["id_kw"] = args.id_column
+    metadata["n_sources"] = histogram.sum()
+    metadata["pix_threshold"] = args.pixel_threshold
+    metadata["urls"] = args.input_paths
+
+    hips_structure = {}
+    temp = [i for i in pixel_map if i is not None]
+    unique_partitions = np.unique(temp, axis=0)
+    for item in unique_partitions:
+        order = int(item[0])
```

Comment:
nitpick: these are already ints, aren't they?

Reply:
They're int64, which the dictionary isn't happy with as a key.